### ISSUE TITLE
Pass Git submodules when creating Bots

### ISSUE THREAD

**Issue by atljeremy** [czechboy0/Buildasaur#270]
I just tried setting up a Buildasaur server with a Bitbucket repo. This Bitbucket repo has a couple Github submodules and it seems the submodules were not cloned in the Xcode Server Bot that Buildasaur created. How do I tell Buildasaur to clone a repo's submodule dependencies?

**Comment by czechboy0** [comment]
Interesting, if Xcode server didn't do that, it's probably something that should be included in the blueprint (but isn't at the moment). Can you please try to create a bot manually and see if those are cloned correctly? If so, it's a missing feature in Buildasaur.

**Comment by atljeremy** [comment]
@czechboy0 I've already created a couple Xcode Bot's manually and each time, it shows the screen in the screenshot below. In this screen it presents me with the repo's submodules and allows me to choose how to authenticate with those submodules (basic auth or private key). After it's been connected, it asks which branch should be used when cloning the submodule, which is what you see in the screenshot. It seems this piece is missing from Buildasaur.

In this screenshot, `HTTPService` and `NotificationQueue` are submodules.

![xcode-server-bot-submodules](https://cloud.githubusercontent.com/assets/1019253/15276747/d55e2676-1abe-11e6-8d14-d604e6dd5d41.png)

**Comment by czechboy0** [comment]
Yeah, it's missing from Buildasaur. Shouldn't be too hard to add, I won't have time to do it in the foreseeable future, I'm afraid. Would happily accept a PR though.

**Comment by mike011** [comment]
Anybody know a workaround? Trying to pull the submodule in a trigger, just seems to hang.

**Comment by czechboy0** [comment]
Yeah, I'd probably try the same. Although I have no previous experience with using submodules with XCS, sorry.

**Comment by krin-san** [comment]
Same bug as in https://github.com/lionheart/openradar-mirror/issues/8030
Waiting for any response in https://forums.developer.apple.com/thread/52357

UPDATE: Can't trust Apple to fix their endless issues. Falling back to CocoaPods solution.
